src/components/ContactList/ContactList.tsx

```diff
@@ -1,16 +1,21 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import styles from './ContactList.module.css';
 import { useDispatch, useSelector } from 'react-redux';
 import { deleteContact, getContacts } from '../../redux/actions';
 import { deleteContactService } from '../../serviсes/deleteContactService';
 import * as phonebookSelectors from '../../redux/selectors';
+import type { AppDispatch } from '../../redux/types'
+import type { ContactData } from '../../types/ContactData';
 
-export default function ContactList({ contactsData }) {
-  const dispatch = useDispatch();
+interface ContactListProps {
+  contactsData: ContactData[]
+}
+
+export default function ContactList({ contactsData }: ContactListProps) {
+  const dispatch = useDispatch<AppDispatch>();
   const error = useSelector(phonebookSelectors.error);
```

Comment:
1. Why types for useSelector have not been provided?
2. State as parameter should be provided

Reply:
Type for useSelector in ContactList has been provided.